lib/saluki-components/src/sources/heartbeat/mod.rs

```diff
@@ -0,0 +1,94 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_context::Context;
+use saluki_core::components::{sources::*, ComponentContext};
+use saluki_core::topology::OutputDefinition;
+use saluki_error::GenericError;
+use saluki_event::{metric::Metric, DataType, Event};
+use tokio::{select, time::interval};
+use tracing::{debug, error};
+
+/// Heartbeat source.
+///
+/// Emits a "heartbeat" metric on a configurable interval.
+#[derive(Clone, Debug)]
+pub struct HeartbeatConfiguration {
+    /// Interval for heartbeat metrics in seconds
+    pub heartbeat_interval_secs: u64,
+}
+
+impl Default for HeartbeatConfiguration {
+    fn default() -> Self {
+        Self {
+            heartbeat_interval_secs: 10,
+        }
+    }
+}
+
+/// Source that doesn't produce any events or optionally emits heartbeat metrics.
+struct HeartbeatSource {
+    heartbeat_interval_secs: u64,
+}
+
+#[async_trait]
+impl Source for HeartbeatSource {
+    async fn run(self: Box<Self>, mut context: SourceContext) -> Result<(), GenericError> {
+        let mut global_shutdown = context.take_shutdown_handle();
+        let mut health = context.take_health_handle();
+        let mut tick_interval = interval(Duration::from_secs(self.heartbeat_interval_secs));
+
+        health.mark_ready();
+        debug!("Heartbeat source started.");
+
+        loop {
+            select! {
+                _ = &mut global_shutdown => {
+                    debug!("Received shutdown signal.");
+                    break;
+                },
+                _ = health.live() => continue,
+                _ = tick_interval.tick() => {
+                    // Create a simple heartbeat metric
+                    let metric_context = Context::from_static_name("none_source.heartbeat");
+                    let metric = Metric::gauge(metric_context, 1.0);
+
+                    let events = vec![Event::Metric(metric)];
+                    if let Err(e) = context.forwarder().forward(events).await {
+                        error!(error = %e, "Failed to forward heartbeat event.");
+                    } else {
+                        debug!("Emitted heartbeat metric.");
```

Comment:
```suggestion
                    let mut buffered_forwarder = context.forwarder().buffered().expect("default output must always exist");
                    if let Err(e) = buffered_forwarder.push(Event::Metric(metric)).await {
                        error!(error = %e, "Failed to forward event.");
                    } else {
                        if let Err(e) = buffered_forwarder.flush().await {
                            error!(error = %e, "Failed to forward events.");
                        } else {
                            debug!("Emitted heartbeat metric.");
                        }
                    }
```

This will be a little more efficient since we skip allocating the `Vec<Event>` and just use `BufferedForwarder<'a, _>` to directly write the event into an event buffer.

(The code is a little verbose, so I opened https://github.com/DataDog/saluki/issues/621 to provide an easier approach.)

Reply:
Awesome. Thanks for the tip. Still getting familiar with Rust and any feedback is very welcome